src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1633,3 +1635,27 @@ def _validate_vmss_terminate_notification(cmd, namespace):  # pylint: disable=un
     """
     if namespace.terminate_notification_time is not None:
         namespace.terminate_notification_time = 'PT' + namespace.terminate_notification_time + 'M'
+
+
+def _validate_vmss_create_automatic_repairs(cmd, namespace):  # pylint: disable=unused-argument
+    if namespace.automatic_repairs_grace_period is not None:
+        if namespace.load_balancer is None or namespace.health_probe is None:
+            raise CLIError("usage error: --load-balancer and --health-probe are required "
+                           "when create vmss with automatic repairs")
+    _validate_vmss_automatic_repairs(cmd, namespace)
+
+
+def _validate_vmss_update_automatic_repairs(cmd, namespace):  # pylint: disable=unused-argument
+    if namespace.enable_automatic_repairs is False and namespace.automatic_repairs_grace_period is not None:
+        raise CLIError("usage error: please enable --enable-automatic-repairs")
+    if namespace.enable_automatic_repairs is True and namespace.automatic_repairs_grace_period is None:
```

Comment:
Is it necessary? Based on your description, I think we can drop this check here.

Reply:
you mean we leave this validation to service side ?

Reply:
default value 30? maybe you can omit this validation

Reply:
per discussed offline, let's just keep this validation.